test/service/test_manifest_async.py

```diff
@@ -182,74 +184,101 @@ def test(self, mock_uuid, mock_helper):
-                            mock_helper.start_execution.assert_called_once_with(
-                                state_machine_name,
-                                execution_id,
-                                execution_input=dict(format_=format_.value,
-                                                     catalog=self.catalog,
-                                                     filters=filters,
-                                                     object_key=self.object_key,
-                                                     partition=dict(index=0,
-                                                                    is_last=False,
-                                                                    file_name=None,
-                                                                    multipart_upload_id=None,
-                                                                    part_etags=None,
-                                                                    page_index=None,
-                                                                    is_last_page=None,
-                                                                    search_after=None))
-                            )
-                            mock_helper.describe_execution.assert_not_called()
-                            mock_helper.reset_mock()
-                            mock_helper.describe_execution.return_value = {'status': 'RUNNING'}
-                            params = None
-                        elif i == 1:
-                            mock_helper.start_execution.assert_not_called()
-                            mock_helper.describe_execution.assert_called_once()
-                            mock_helper.reset_mock()
-                            # simulate absence of output due eventual consistency
-                            mock_helper.describe_execution.return_value = {'status': 'SUCCEEDED'}
-                        elif i == 2:
-                            mock_helper.start_execution.assert_not_called()
-                            mock_helper.describe_execution.assert_called_once()
-                            mock_helper.reset_mock()
-                            mock_helper.describe_execution.return_value = {
-                                'status': 'SUCCEEDED',
-                                'output': json.dumps(
-                                    {
-                                        'manifest': {
-                                            'location': manifest_url.url,
-                                            'was_cached': False,
-                                            'format_': format_.value,
-                                            'catalog': self.catalog,
-                                            'filters': filters,
-                                            'object_key': object_key
-                                        }
-                                    }
+                    partition = dict(index=0,
```

Comment:
Probably better to use the partition constructor and to_json instead of dict and from_json